### PR TITLE
HTS-861: countryCode parsing is done in the frontend

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk.gov.hmrc.helptosaveproxy/models/NSIUserInfo.scala [hunk 1]
```diff
@@ -92,7 +89,7 @@ object NSIUserInfo {
           c.address4.map(_.cleanupSpecialCharacters),
           c.address5.map(_.cleanupSpecialCharacters),
           c.postcode.cleanupSpecialCharacters.removeAllSpaces,
-          c.countryCode.map(_.cleanupSpecialCharacters.removeAllSpaces).filter(_.toLowerCase =!= "other").map(_.take(2)),
+          c.countryCode,
```
Comment (AndyHWChung) [inline app/uk.gov.hmrc.helptosaveproxy/models/NSIUserInfo.scala]: we need to have this logic sitting in the FE somewhere - possibly here: https://github.com/hmrc/help-to-save-frontend/blob/master/app/uk/gov/hmrc/helptosavefrontend/models/userinfo/Address.scala#L39 or somewhere around here: https://github.com/hmrc/help-to-save-frontend/blob/master/app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveAuth.scala#L157

Comment (sureshhmrc) [reply app/uk.gov.hmrc.helptosaveproxy/models/NSIUserInfo.scala]: its already there here: 
```
https://github.com/hmrc/help-to-save-frontend/blob/master/app/uk/gov/hmrc/helptosavefrontend/models/userinfo/NSIUserInfo.scala#L82
```

Comment (AndyHWChung) [reply app/uk.gov.hmrc.helptosaveproxy/models/NSIUserInfo.scala]: ah, I missed that - thanks for finding it

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**